DataGenerator/DataGenerator.py

```diff
@@ -18,180 +19,253 @@
 from Utils.GenerateSmoothLabel import get_smoothed_label_distribution
 from sklearn.preprocessing import StandardScaler
 import xnat
+import SimpleITK as sitk
+from monai.data import image_reader
+from monai.transforms import LoadImage, LoadImaged
+from numpy import array
+from scipy.interpolate import RegularGridInterpolator as rgi
+from scipy.ndimage import map_coordinates
 
 class DataGenerator(torch.utils.data.Dataset):
-    def __init__(self, PatientList, label, config, keys, inference=False, n_norm = None, c_norm = None, transform=None, target_transform = None):
+    def __init__(self, PatientList, config, keys, transform=None, **kwargs):
         super().__init__()
-        self.keys             = list(keys)
-        self.transform        = transform
-        self.target_transform = target_transform
-        self.label            = label
-        self.inference        = inference
-        self.PatientList      = PatientList
-        self.n_norm = n_norm
-        self.c_norm = c_norm
+        self.transform = transform
+        self.keys = keys
+        self.inference = kwargs['inference']
+        self.PatientList = PatientList
+        # self.n_norm = kwargs['numerical_norm']
+        # self.c_norm = kwargs['category_norm']
         self.config = config
 
     def __len__(self):
-        return int(self.mastersheet.shape[0])
+        return int(len(self.PatientList))
 
     def __getitem__(self, id):
 
         datadict = {}
         roiSize = [10, 40, 40]
-        AnatomyPath = PatientList[i].scan.experiment['CT'].path()
-        
-        #label    = self.mastersheet[self.label].iloc[id]
+        patient_id = self.PatientList[id].label
+        ScanPath = self.config['DATA']['DataFolder'] + patient_id + '\\' + patient_id + '\\' + 'scans\\'
+        subfolder_list = os.listdir(ScanPath)
+        reader = image_reader.ITKReader()
+        label = self.PatientList[id].fields[self.config['DATA']['target']]
+        CT_match_folder = [match for match in subfolder_list if "CT" in match]
+        full_CT_path = ScanPath + CT_match_folder[0] + '\\resources\\DICOM\\files\\'
+        dicom_files = os.listdir(full_CT_path)
+        correct_Origin = reader.read(full_CT_path + dicom_files[0])
+        itkObj = reader.read(full_CT_path)
+        itkObj.SetOrigin(correct_Origin.GetOrigin())
         # Get the mask of PTV
         if "Dose" in self.keys or "Anatomy" in self.keys:
             if self.config['DATA']['Use_mask']:
-                path = self.mastersheet["CTPath"].iloc[id]
-                mask = path[0:-16] + 'structs\PTV.nrrd'
-                mask_img = LoadImg(mask)
+                RT_match_folder = [match for match in subfolder_list if "Structs" in match]
```

Comment:
Both for CT and structs, there shouldn't be a "matching parameter". We have to work with Olivia to have consistency between the naming of the different modalities and call them directly.